ggshield/scan/repo.py

```diff
@@ -102,21 +147,26 @@ def scan_commit_range(
     :param verbose: Display successful scan's message
     """
 
+    commits_batch = get_commits_by_batch(
+        commit_list=[
+            Commit(sha=sha, exclusion_regexes=exclusion_regexes) for sha in commit_list
+        ]
```

Comment:
`commit_list` argument can be turned into a generator to avoid creating a list:

```suggestion
        commit_list=(
            Commit(sha=sha, exclusion_regexes=exclusion_regexes) for sha in commit_list
        )
```